### PR TITLE
[fix](routine load) reset Kafka progress cache when routine load job topic change

### PR DESCRIPTION
**Pull request by sollhui**
When change routine load job topic from test_topic_before to test_topic_after by 
```
ALTER ROUTINE LOAD FOR test_topic_change FROM KAFKA("kafka_topic" = "test_topic_after");
```
(test_topic_before has 5 rows and test_topic_after has 1 rows)

Exception happened, which cannot consume any data:
```
2024-07-29 15:57:28,122 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,123 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,125 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,126 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,128 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,129 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,131 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,133 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,134 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,136 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
2024-07-29 15:57:28,137 WARN (Routine load task scheduler|55) [KafkaRoutineLoadJob.hasMoreDataToConsume():792] Kafka offset fallback. partition: 0, cache offset: 5 get latest of
fset: 1, task 16656914-ba0a-465d-8e79-8252b423b0fc, job 16615
```

It is necessary to reset Kafka progress cache when routine load job topic change.

### CODE REVIEW

**Comment by sollhui** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by sollhui** [comment]
run buildall

**Review by liaoxin01 — DISMISSED** [review]
LGTM

**Comment by sollhui** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java [hunk 1]
```diff
@@ -713,22 +713,35 @@ private void modifyPropertiesInternal(Map<String, String> jobProperties,
...
+                    resetCloudProgress();
+                }
                 this.topic = dataSourceProperties.getTopic();
+                this.progress = new KafkaProgress();
+            }
+
+            // modify partition offset
+            if (!kafkaPartitionOffsets.isEmpty()) {
+                // we can only modify the partition that is being consumed
+                ((KafkaProgress) progress).modifyOffset(kafkaPartitionOffsets);
```
Comment (liaoxin01) [inline fe/fe-core/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java:739]: Do we need to modify the progress  of meta service in cloud mode?

Comment (sollhui) [reply fe/fe-core/src/main/java/org/apache/doris/load/routineload/KafkaRoutineLoadJob.java:739]: We need, and will fix it in next pr

File: cloud/src/meta-service/meta_service_txn.cpp [hunk 2]
```diff
@@ -644,6 +644,58 @@ void MetaServiceImpl::get_rl_task_commit_attach(::google::protobuf::RpcControlle
...
+
+    int64_t db_id = request->db_id();
+    int64_t job_id = request->job_id();
+    std::string rl_progress_key;
+    std::string rl_progress_val;
+    RLJobProgressKeyInfo rl_progress_key_info {instance_id, db_id, job_id};
+    rl_job_progress_key_info(rl_progress_key_info, &rl_progress_key);
+    txn->remove(rl_progress_key);
+    err = txn->commit();
+    if (err == TxnErrorCode::TXN_KEY_NOT_FOUND) {
```
[the remark this answers is not in the document]
Comment (sollhui) [reply cloud/src/meta-service/meta_service_txn.cpp:685]: Handle in 
```
 if (response.getStatus().getCode() == Cloud.MetaServiceCode.ROUTINE_LOAD_PROGRESS_NOT_FOUND) {
                    LOG.warn("not found routine load progress, response: {}", response);
                    return;
                }
```

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Review by liaoxin01 — DISMISSED** [review]
LGTM

**Comment by sollhui** [comment]
run buildall

**Comment by sollhui** [comment]
run buildall

**Review by liaoxin01 — APPROVED** [review]

**Review by XuJianxu — APPROVED** [review]
LGTM

**Pull request merged**